### PR TITLE
WIP - CAMEL-10287: Infinispan RoutePolicy to have one route being master, and others as slaves

### PR DESCRIPTION
**Pull request by lburgazzoli**
@tristantarrant may you check if I missed something on infinispan side

### CODE REVIEW

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+
+        @Override
+        public void run() {
+            if (!isRunAllowed() || !InfinispanRoutePolicy.this.isRunAllowed()) {
+                return;
+            }
+
+            if (isLeader()) {
+                // I'm still the leader, so refresh the key so it does not expire.
+                if (!cache.replace(lockKey, lockKey, lockKey, lifespan, lifespanTimeUnit)) {
```
Comment (pruivo) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java]: why replacing ```lockKey``` by ```lockKey```? shouldn't be ```lockValue``` by ```lockValue```?

Comment (lburgazzoli) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java]: yeah it is definitively a mistake

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+        public void run() {
+            if (!isRunAllowed() || !InfinispanRoutePolicy.this.isRunAllowed()) {
+                return;
+            }
+
+            if (isLeader() && version != null) {
+                LOGGER.debug("Lock refresh key={} with version={}", lockKey, version);
+
+                // I'm still the leader, so refresh the key so it does not expire.
+                if (!cache.replaceWithVersion(lockKey, lockKey, version, (int)lifespanTimeUnit.toSeconds(lifespan))) {
```
Comment (pruivo) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java]: same as above. the new value should be ```lockValue```

File: components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+    //
+    // *************************************************************************
+
+    protected void setLeader(boolean isLeader) {
+        if (isLeader && leader.compareAndSet(false, isLeader)) {
+            LOGGER.info("Leadership taken (map={}, key={}, val={})", lockMapName, lockKey, lockValue);
+
+            startAllStoppedConsumers();
+        } else {
+            if (!leader.getAndSet(isLeader) && isLeader) {
```
Comment (pruivo) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java]: can someone review this condition? I don't think the log is printed correctly. 
If I understand correctly, it should be ```leader.getAndSet(isLeader) && !isLeader```. ```leader``` was ```true``` and it was set to ```false```.

Comment (lburgazzoli) [inline components/camel-infinispan/src/main/java/org/apache/camel/component/infinispan/policy/InfinispanRoutePolicy.java]: Correct going to fix it

**Comment by lburgazzoli** [comment]
@pruivo I've fixed the issues you highlighted

**Comment by pruivo** [comment]
@lburgazzoli no more comments from me. It looks good.

**Comment by lburgazzoli** [comment]
Merged via dbd68347b29ce7813262ed2a63fb31e36e94a034

**Pull request closed without merging**